src/Renderer.cpp

```diff
@@ -20,10 +19,36 @@ namespace ccm
 		setRenderColor(Colors::Black);
 		SDL_RenderClear(m_renderer);
 	}
+
 	void Renderer::draw(const Object& obj)
 	{
-		setRenderColor(obj.color);
-		SDL_RenderFillRect(m_renderer, &obj.rect);
+		switch(obj.getTypeIndex())
+		{
+		case (unsigned int)Object::variantIndex::Color:
```

Comment:
The variant should already know which type it holds. I don't think there's a need for a `variantIndex`. Looks like you're trying to do what `std::visit` accomplishes.